docs/Reference/middlewareExtensibility.md

```diff
@@ -0,0 +1,57 @@
+--8<-- "References/abbreviations.md"
+
+Middleware extensibility allows developers to implement new middleware handlers to both MPS or RPS. By adding custom functions, MPS and RPS will process and load these during server startup. The `loadCustomMiddleware` function that executes on startup can be found in `mps/src/server/webserver.ts` or `rps/src/index.ts`.
+
+Example use-cases:
+
+- Handling Custom Authentication Tokens
+- Adding Trace IDs
+- Debugging Requests
+
+To demonstrate the execution, we'll use an example.  Let's say a custom auth token handler was implemented across all API endpoints.  When a call is then made against an API endpoint, the custom handler will execute first.  This handler might process the token.  After executing the custom function and calling `next()`, API execution will then continue as normal.
+
+## Add a Custom Middleware Function
+
+To add a new function, create a new typescript file in `/src/middleware/custom/`. An `example.ts` file is already provided in this directory.
+
+The file **must have two key parts** in order to successfully load:
+
+1. Must be exported as a default. **Only the single, default function will be what is loaded into MPS or RPS.** Additional functions that need to be loaded will need their own separate `.ts` files.
```

Comment:
Should we add?... "The desired function" in front of ..."must be exported as default"